core/dmux.go

```diff
@@ -311,7 +378,172 @@ func simpleSetup(size, qsize int, sink Sink) ([]chan interface{}, *sync.WaitGrou
 		go func(index int) {
 			sk := sink.Clone()
 			for msg := range ch[index] {
-				sk.Consume(msg)
+				sk.Consume(msg, math.MaxInt32)
+			}
+			wg.Done()
+		}(i)
+	}
+	return ch, wg
+}
+
+func simpleSetupWithSideline(size, qsize int, sink Sink, sideline Sideline, sidelineImpl interface{}) ([]chan interface{}, *sync.WaitGroup) {
+	wg := new(sync.WaitGroup)
+	wg.Add(size)
+	ch := make([]chan interface{}, size)
+	sidelineChannel := make([]chan ChannelObject, size)
+	sinkChannel := make([]chan ChannelObject, size)
+
+	for i := 0; i < size; i++ {
+		sinkChannel[i] = make(chan ChannelObject, qsize)
+		go func(index int) {
+			sk := sink.Clone()
+			for channelObject := range sinkChannel[index] {
+				log.Printf("Inside Sink channel ")
+				for {
+					consumeError := sk.Consume(channelObject.Msg, sideline.Retries)
+					if consumeError == nil {
+						break
+					}
+					if consumeError != nil && consumeError.Error() == "exceeded retries" {
```

Comment:
`consumeError != nil` is obsolete here. should the decision to sideline or not be part of sideline provider itself?

Reply:
sideline provider only does the sideline of a message and check if a message with same key has already been sidelined.
In this code flow it is calling sink consume method, so there is involvement of sideline provider.
let me know if i am missing anything.